http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala

```diff
@@ -179,6 +179,37 @@ class HttpSamDAO[F[_]](httpClient: Client[F],
         }
     } yield ()
 
+  override def createResourceV2[R](resource: R,
+                                   creatorEmail: WorkbenchEmail,
+                                   cloudContext: CloudContext,
+                                   userInfo: UserInfo
+  )(implicit sr: SamResource[R], ev: Ask[F, TraceId]): F[Unit] = for {
+    traceId <- ev.ask
+    authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
```

Comment:
Noting there is a slight change compared to `createResource`. In `createResource` we request the pet key based on the `creatorEmail`, and make the Sam calls as the pet. In this method we call Sam with the user token directly.

I think using the token directly is probably fine for `createResourceV2`, since it comes from Front Leo (where we have the token). However we'll probably need to call `deleteResourceV2` from Back Leo once deleteApp is [implemented](https://broadworkbench.atlassian.net/browse/TOAZ-230). 

I think we can probably start with this though, and revisit as part of adding deleteApp (I think there are a couple solutions..)

Reply:
Also, is the only difference between `createResource` and `createResourceV2` the fact that the former uses a GCP pet to make the call?

Maybe we could name the `SamDAO` methods based on that difference in behavior:

```
def createResourceWithGcpPet(resource: R, creatorEmail: WorkbenchEmail, googleProject: GoogleProject)

def createResourceAsUser(resource: R, userInfo: UserInfo)
```

Reply:
I'll keep the implementation as is!

I like the name adjustments, I'll do that.